jni/test/jni_array_test.dart

```diff
@@ -0,0 +1,212 @@
+import 'dart:io';
+
+import 'package:jni/jni.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // Don't forget to initialize JNI.
+  if (!Platform.isAndroid) {
+    try {
+      Jni.spawn(dylibDir: "build/jni_libs", jvmOptions: ["-Xmx128m"]);
+    } on JvmExistsException catch (_) {
+      // TODO(#51): Support destroying and reinstantiating JVM.
+    }
+  }
+  test("Java boolean array", () {
+    final array = Jni.newBooleanArray(3);
+    expect(array.length, 3);
+    array[0] = true;
+    array[1] = false;
+    array[2] = false;
+    expect(array[0], true);
+    expect(array[1], false);
+    expect(array[2], false);
+    array.setRange(0, 3, [false, true, true, true], 1);
+    expect(array[0], true);
+    expect(array[1], true);
+    expect(array[2], true);
+    expect(() {
+      array[-1] = false;
+    }, throwsRangeError);
+    expect(() {
+      array[3] = false;
+    }, throwsRangeError);
+    array.delete();
+  });
+  test("Java char array", () {
+    final array = Jni.newCharArray(3);
+    expect(array.length, 3);
+    array[0] = '1';
+    array[1] = '2';
+    array[2] = '3';
+    expect(array[0], '1');
+    expect(array[1], '2');
+    expect(array[2], '3');
+    array.setRange(0, 3, ['4', '5', '6', '7'], 1);
+    expect(array[0], '5');
+    expect(array[1], '6');
+    expect(array[2], '7');
+    expect(() {
+      array[-1] = '4';
+    }, throwsRangeError);
+    expect(() {
+      array[3] = '4';
+    }, throwsRangeError);
+    array.delete();
+  });
+  test("Java byte array", () {
+    final array = Jni.newByteArray(3);
+    expect(array.length, 3);
+    array[0] = 1;
+    array[1] = 2;
+    array[2] = 3;
+    expect(array[0], 1);
+    expect(array[1], 2);
+    expect(array[2], 3);
+    array.setRange(0, 3, [4, 5, 6, 7], 1);
+    expect(array[0], 5);
+    expect(array[1], 6);
+    expect(array[2], 7);
+    expect(() {
+      array[-1] = 4;
+    }, throwsRangeError);
+    expect(() {
+      array[3] = 4;
+    }, throwsRangeError);
+    array.delete();
+  });
+  test("Java short array", () {
```

Comment:
There seems to be some redundancy here. Can this somehow be extracted into an interface so that the common code can be factored into a function?

Reply:
For now it helped me identify a bug I had. We have an extension for each primitive type, it's difficult to make a generic common code that uses `[]` and `[]=` for the same reason that we couldn't have it without multiple extensions.

Reply:
Yeah it seems extensions can't implement interfaces nevermind.